src/web/src/pages/room/VotingRoomPage/index.tsx

```diff
@@ -47,52 +51,144 @@ function QuestionVoter({ data }: { data: VotingPageState }) {
+    schema: questionResponse,
+    defaultValues: lastVote,
+  });
 
-    // Randomly reorder indexes array
-    for (let i = indexes.length - 1; i > 0; i--) {
-      const j = Math.floor(Math.random() * (i + 1));
-      [indexes[i], indexes[j]] = [indexes[j], indexes[i]];
+  const onSubmit = handleSubmit(async (data) => {
+    switch (data.type) {
+      case 'Abstain':
+        castVote({ type: 'Abstain' });
+        break;
+      case 'SingleVote':
+        castVote({
+          type: 'SingleVote',
+          candidateId: data.candidateId,
+        });
+        break;
+      case 'PreferentialVote':
+        castVote({
+          candidateIds: data.candidateIds,
+          type: 'PreferentialVote',
+        });
+        break;
     }
+  });
+
+  // const reorder = useMemo(() => {
+  //   const indexes = Array.from({ length: data.question.candidates.length }, (_, i) => i);
+
+  //   // Randomly reorder indexes array
+  //   for (let i = indexes.length - 1; i > 0; i--) {
+  //     const j = Math.floor(Math.random() * (i + 1));
+  //     [indexes[i], indexes[j]] = [indexes[j], indexes[i]];
+  //   }
+
+  //   return indexes;
+  // }, [data.question.questionId]);
+
+  // const candidatesReordered = useMemo(() => question.candidates.map((_, i) => question.candidates[reorder[i]]), [reorder]);
+
+  // const [candidateOrder, setCandidateOrder] = useState<VotingCandidate[]>(candidatesReordered);
+
+  // useEffect(() => {
+  //   setCandidateOrder(candidatesReordered);
+  // }, [candidatesReordered]);
 
-    return indexes;
-  }, [data.question.questionId]);
+  // const debounced = useDebounceCallback((votingCandidates: VotingCandidate[]) => {
+  //   castVote({
+  //     type: 'PreferentialVote',
+  //     candidateIds: votingCandidates.map(candidate => candidate.id),
+  //   });
+  // }, 500);
 
-  const candidatesReordered = question.candidates.map((_, i) => question.candidates[reorder[i]]);
+  // const onReorder = (unknownVotingCandidates: unknown[]) => {
+  //   const votingCandidates = unknownVotingCandidates as VotingCandidate[];
+  //   debounced(votingCandidates);
+
+  //   setCandidateOrder(votingCandidates);
+  // };
+
+  useEffect(() => {
+    if (lastVote) {
+      reset(lastVote);
+    }
+  }, [lastVote]);
 
   return (
-    <div className="flex flex-col items-center gap-6 w-full">
+    <form className="flex flex-col items-center gap-6 w-full">
       <Question>{question.question}</Question>
-      <div className="flex gap-4 flex-wrap flex-col sm:flex-row items-stretch sm:items-center justify-center w-full">
-        {candidatesReordered.map(candidate => (
-          <Button
-            className={twMerge(
-              lastVote?.type === 'SingleVote' && lastVote.candidateId === candidate.id && 'btn-accent',
+      <div className="flex gap-4 flex-wrap flex-col items-stretch justify-center w-full">
+        {question.details.type === 'SingleVote'
+          ? question.candidates.map(candidate => (
+            <Controller
```

Comment:
I'm not sure how this works, but every time I press a candidate in single vote mode it reloads the entire page

Reply:
fixed